src/peer_manager.rs

```diff
@@ -227,6 +142,14 @@ impl Peer {
         self.timestamp.elapsed() >= Duration::from_secs(timeout)
     }
 
+    /// Returns whether the peer is in the `Connecting` state.
+    pub fn is_connecting(&self) -> bool {
+        match self.state {
```

Comment:
For these kind of situations, I'm in favour of deriving `PartialEq` for `PeerState` and simplifying this code except if there is a good reason not to do so.